contracts/SwapEthWrapper.sol

```diff
@@ -0,0 +1,298 @@
+     * in position of WETH9.
+     * @param amounts the amounts of each token to add, in their native precision
+     * @param minToMint the minimum LP tokens adding this amount of liquidity
+     * should mint, otherwise revert. Handy for front-running mitigation
+     * @param deadline latest timestamp to accept this transaction
+     * @return amount of LP token user minted and received
+     */
+    function addLiquidity(
+        uint256[] memory amounts,
+        uint256 minToMint,
+        uint256 deadline
+    ) external payable returns (uint256) {
+        // If using ETH, deposit them to WETH.
+        if (msg.value > 0) {
+            IWETH9(weth).deposit{value: msg.value}();
+        }
+        require(msg.value == amounts[wethIndex], "INCORRECT_MSG_VALUE");
+        // Go through amounts array and transfer respective tokens to this contract.
+        for (uint256 i = 0; i < amounts.length; i++) {
+            uint256 amount = amounts[i];
+            if (i != wethIndex && amount > 0) {
+                pooledTokens[i].safeTransferFrom(
+                    msg.sender,
+                    address(this),
+                    amount
+                );
+            }
+        }
+        // Add the assets to the pool
+        uint256 lpTokenAmount =
+            swapInstance.addLiquidity(amounts, minToMint, deadline);
+        // Send the LPToken to msg.sender
+        IERC20(address(lpToken)).safeTransfer(msg.sender, lpTokenAmount);
+        return lpTokenAmount;
+    }
+
+    /**
+     * @notice Burn LP tokens to remove liquidity from the pool.
+     * @dev Liquidity can always be removed, even when the pool is paused. Caller
+     * will receive ETH instead of WETH9.
+     * @param amount the amount of LP tokens to burn
+     * @param minAmounts the minimum amounts of each token in the pool
+     *        acceptable for this burn. Useful as a front-running mitigation
+     * @param deadline latest timestamp to accept this transaction
+     * @return amounts of tokens user received
+     */
+    function removeLiquidity(
+        uint256 amount,
+        uint256[] calldata minAmounts,
+        uint256 deadline
+    ) external returns (uint256[] memory) {
+        // Transfer LPToken from msg.sender to this contract.
+        IERC20(address(lpToken)).safeTransferFrom(
+            msg.sender,
+            address(this),
+            amount
+        );
+        // Remove liquidity
+        uint256[] memory amounts =
+            swapInstance.removeLiquidity(amount, minAmounts, deadline);
+        // Send the tokens back to the user
+        for (uint256 i = 0; i < amounts.length; i++) {
+            if (i != wethIndex) {
+                pooledTokens[i].safeTransfer(msg.sender, amounts[i]);
+            } else {
+                IWETH9(weth).withdraw(amounts[i]);
+                // slither-disable-next-line arbitrary-send
+                (bool success, ) = msg.sender.call{value: amounts[i]}("");
```

Comment:
Why performing the send like this instead of a normal transfer?

Reply:
Not sure if this is a legit concern as of today but in case the gas requirements for certain smart contract interactions change, `.transfer` _could_ fail. 

https://consensys.net/diligence/blog/2019/09/stop-using-soliditys-transfer-now/